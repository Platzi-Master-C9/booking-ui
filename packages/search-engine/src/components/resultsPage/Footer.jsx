import React from 'react';

export default function Footer() {
  return (
    <footer className="footer__container">
      <div className="footer__wrapper">
        <a className='link' href='./'>Sobre Nosotros</a>
        <a className='link' href='./'>Prop√≥sito del proyecto</a>
        <a className='link' href='./'>Redes Sociales</a>
      </div>
    </footer>
  )
}