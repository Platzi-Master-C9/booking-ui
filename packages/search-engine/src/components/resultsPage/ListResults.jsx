import React from 'react';
// import { Card } from '../Card';
import './styles/ListResults.css'

function ListResults({ places }) {
  return <React.Fragment>
    <section className='list__results'>
      { places.map(card => {
        return <div className="results__card" key={card.uuid}> {card.city} </div>
      })}
    </section>
  </React.Fragment>
}

export { ListResults };
