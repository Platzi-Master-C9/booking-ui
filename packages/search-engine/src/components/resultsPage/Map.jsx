import React from 'react';

export default function Map() {
  return <div className="map__container"><img/></div>
}