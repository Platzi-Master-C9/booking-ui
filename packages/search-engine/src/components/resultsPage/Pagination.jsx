import React from 'react';
<<<<<<< HEAD
import './styles/Pagination.css'
=======
>>>>>>> Layout the search results page #41

export default function Pagination({currentPage, prevHandler, nextHandler, TotalPages}) {
  return <div className='paging__container'>
    <button className='icon icon--left' onClick={prevHandler}></button>
<<<<<<< HEAD
    <div>
      <span className='paging paging-start'> {currentPage + 1} </span>
      <span className='paging'> de </span>
      <span className='paging'> {TotalPages} </span>
    </div>
=======
    <div>{currentPage + 1}</div>
    <div>{currentPage + 2}</div>
    <div>{currentPage + 3}</div>
    <div>...</div>
    <div>{TotalPages}</div>
>>>>>>> Layout the search results page #41
    <button className='icon icon--right' onClick={nextHandler}></button>
  </div>
}
