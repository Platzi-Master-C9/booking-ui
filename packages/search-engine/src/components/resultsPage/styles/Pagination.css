:root {
  /* colors */
  --color-font-dark: #000000;
  --color-font-middle: #666666;
  --color-font-light: #FFFFFF;
  --color-font-alt: #1572A1;
  --color-main: #1572A1;

  /* font-size */
  --size-font-l: 36px;
  --size-font-m: 24px;
  --size-font-s: 18px;
}

.paging__container {
  grid-area: pages;
  width: 100%;
  height: 42px;
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 10px;
}

.icon {
  width: 24px;
  height: 24px;
  border: none;
  background-color: inherit;
  background-image: url(../assets/icons8-expand-arrow-96.png);
  background-position: center;
  background-repeat: no-repeat;
  background-size: 24px;
}

.icon--left {
  transform: rotate(90deg);
}

.icon--right {
  transform: rotate(-90deg);
}

.paging {
  display: inline-block;
  width: 40px;
  height: 40px;
  font-size: var(--size-font-s);
  font-weight: 600;
  line-height: 2.4;
  text-align: center;
  color: var(--color-font-dark);
}

.paging-start {
  border-radius: 20px;
  color: var(--color-font-light);
  background-color: var(--color-main);
}
