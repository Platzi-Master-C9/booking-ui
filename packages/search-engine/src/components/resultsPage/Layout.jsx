import React, { useEffect, useState } from "react";

import Header from "../Header"
import { Map } from "../Map"
import Footer from "../Footer"
import { ListResults } from "./ListResults"
import { Pagination } from "./Pagination"
import "./styles/style.css"
import { DATA_API } from "./API/data";
import { getData } from "../../scripts/petitions";
import { ErrorMessage } from "../ErrorMessage";

const PLACES_PER_PAGE = 10;
const key = 15;

const ResultsPage = () => {
  
  const [ allPlaces, setAllPlaces ] = useState([]);

  const [ renderedPlaces, setRenderedPlaces ] = useState([]);
  const [ dataState, setDataState ] = useState({ loading: true, error: false });
  const [ currentPage, setCurrentPage ] = useState(0);
  const [ TotalPages, setTotalPages ] = useState(0);
  const [error, setError] = useState(false);

  useEffect(() => {
    try {
      getData(key).then(res => setAllPlaces(res.data));
      setDataState({ ...dataState, loading: false });
    } catch(error) {
      setDataState({ ...dataState, error: error });
    }
  }, []);

  useEffect(() => {
    setRenderedPlaces([...allPlaces].splice(0, PLACES_PER_PAGE));

  useEffect(() => {
    setPlaces([...allPlaces].splice(0, PLACES_PER_PAGE));
    setTotalPages(Math.ceil(allPlaces.length / PLACES_PER_PAGE));
  }, [allPlaces]);

  const prevHandler = () => {
    const prevPage = currentPage - 1;

    if(prevPage < 0) { return }
    const currentIndex = prevPage * PLACES_PER_PAGE;

    setRenderedPlaces([...allPlaces].splice(currentIndex, PLACES_PER_PAGE));
    setCurrentPage(prevPage);
  }

  const nextHandler = () => {
    const TotalPlaces = allPlaces.length;
    const nextPage = currentPage + 1;
    const currentIndex = nextPage * PLACES_PER_PAGE;

    if(currentIndex >= TotalPlaces) { return }

    setRenderedPlaces([...allPlaces].splice(currentIndex, PLACES_PER_PAGE));
    setCurrentPage(nextPage);
  }

  const handleSubmitFilters = async (minPrice, maxPrice, stars, fullPlace, privateRoom) => {
    const data = await getData('places');
    if(data.error) {
      setError('Ha ocurrido un error al obtener los lugares');
      return;
    };
    setAllPlaces(data);
  }

  return (
    <div className="results__container">
      <Header handleSubmitFilters={handleSubmitFilters} />

      <section className="section__container">
        <Map />

        {!error ? 
          <>
            <ListResults 
              places={renderedPlaces}
              statePlaces={dataState}
            />
            <Pagination 
              TotalPages={totalPages} 
              currentPage={currentPage} 
              prevHandler={prevHandler} 
              nextHandler={nextHandler} 
            />
          </>
          : 
          <ErrorMessage message={error} />
        }
      </section>

      <Footer />
    </div>
  )
}

export { ResultsPage };
