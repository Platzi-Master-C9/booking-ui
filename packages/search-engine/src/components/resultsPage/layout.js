import React, { useState } from "react";

import Header from "../Header"
import Map from "../Map"
import Footer from "../Footer"
import ListResults from "./ListResults"
import Pagination from "./Pagination"
import "./styles/style.css"
import { DATA_API } from "./API/data";
// import getData from "./API/getData";

export const ResultsPage = () => {
// export const ResultsPage = ({ DataFromApi }) => {
  const PLACES_PER_PAGE = 10;
  
  const [ allPlaces, setAllPlaces ] = useState(DATA_API); //debbuging
  // const [ allPlaces, setAllPlaces ] = useState(DataFromApi); //debbuging
  const [ places, setPlaces ] = useState([...allPlaces].splice(0, PLACES_PER_PAGE)); //debbuging
  const [ currentPage, setCurrentPage ] = useState(0);
  const TotalPages = Math.ceil(allPlaces.length / PLACES_PER_PAGE);

  const prevHandler = () => {
    const prevPage = currentPage - 1;

    if(prevPage < 0) { return }
    const currentIndex = prevPage * PLACES_PER_PAGE;

    setPlaces([...allPlaces].splice(currentIndex, PLACES_PER_PAGE));
    setCurrentPage(prevPage);
  }

  const nextHandler = () => {
    const TotalPlaces = allPlaces.length;
    const nextPage = currentPage + 1;
    const currentIndex = nextPage * PLACES_PER_PAGE;

    if(currentIndex === TotalPlaces) { return }

    setPlaces([...allPlaces].splice(currentIndex, PLACES_PER_PAGE));
    setCurrentPage(nextPage);
  }

  return (
    <div className="results__container">
      <Header />

      <section className="section__container">
        <Map />
        <ListResults 
          data={places} 
        />
        <Pagination 
          TotalPages={TotalPages} 
          currentPage={currentPage} 
          prevHandler={prevHandler} 
          nextHandler={nextHandler} 
        />
      </section>

      <Footer />
    </div>
  )
}
