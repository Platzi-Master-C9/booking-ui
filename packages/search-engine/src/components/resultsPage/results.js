import React, { useState, useEffect } from "react";

import Header from "./Header"
import Map from "./Map"
import Footer from "./Footer"
import ListResults from "./ListResults"
import Pagination from "./Pagination"
import "./styles/styles.css"

export const ResultsPage = () => {
  
  const [allPlaces, setAllPlaces] = useState([]);
  
  const GetDataFromApi = () => {
    fetch(process.env.REACT_APP_API_RESULTS)
      .then(response => response.json())
      .then(data => setAllPlaces(data))
      .catch(error => console.error(error))
  }

  useEffect(() => {
    GetDataFromApi();
  },[]);

  console.log('In ResultsPage ', allPlaces)

  const PLACES_PER_PAGE = 10;

  const [ places, setPlaces ] = useState([...allPlaces].splice(0, PLACES_PER_PAGE)); //debbuging
 
  console.log('In places ', places)

  const [ currentPage, setCurrentPage ] = useState(0);
  const TotalPages = Math.ceil(allPlaces.length / PLACES_PER_PAGE);

  const prevHandler = () => {
    const prevPage = currentPage - 1;

    if(prevPage < 0) { return }
    const currentIndex = prevPage * PLACES_PER_PAGE;

    setPlaces([...dataFromApi].splice(currentIndex, PLACES_PER_PAGE));
    setCurrentPage(prevPage);
  }

  const nextHandler = () => {
    const TotalPlaces = dataFromApi.length;
    const nextPage = currentPage + 1;
    const currentIndex = nextPage * PLACES_PER_PAGE;

    if(currentIndex === TotalPlaces) { return }

    setPlaces([...dataFromApi].splice(currentIndex, PLACES_PER_PAGE));
    setCurrentPage(nextPage);
  }

  return <div className="results__container">
    <Header />

    <section className="section__container">
      <Map />
      <ListResults data={places} />
      <Pagination TotalPages={TotalPages} currentPage={currentPage} prevHandler={prevHandler} nextHandler={nextHandler} />
    </section>

    <Footer />
  </div>
}
