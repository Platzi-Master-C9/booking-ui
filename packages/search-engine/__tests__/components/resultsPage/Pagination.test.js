import React from 'react';
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer';

import { Pagination } from '../../../src/components/resultsPage/Pagination';

describe('<Pagination />', () => {
  const Mock = {
    currentPage: 1,
    prevHandler: jest.fn(),
    nextHandler: jest.fn(),
    TotalPages: 30
  };

  it('Render the pagination component', () => {
    const component = shallow(
      <Pagination
       currentPage={Mock.currentPage}
       prevHandler={Mock.prevHandler}
       nextHandler={Mock.nextHandler}
       TotalPages={Mock.TotalPages}
      />
    );

    expect(component.length).toEqual(1);
    expect(component.find('.paging').length).toEqual(3);
    
    component.find('.icon--right').simulate('click');
    expect(Mock.nextHandler).toHaveBeenCalledTimes(1);
  });
});

describe('Pagination Snapshot', () => {
  it('compounding the UI of the component', () => {
    const pagination = create(<Pagination />);
    expect(pagination.toJSON()).toMatchSnapshot();
  });
});
