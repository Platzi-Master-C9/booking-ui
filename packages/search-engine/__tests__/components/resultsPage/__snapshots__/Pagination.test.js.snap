// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Snapshot compounding the UI of the component 1`] = `
<div
  className="paging__container"
>
  <button
    className="icon icon--left"
  />
  <div>
    <span
      className="paging paging-start"
    >
       
      NaN
       
    </span>
    <span
      className="paging"
    >
       de 
    </span>
    <span
      className="paging"
    >
       
       
    </span>
  </div>
  <button
    className="icon icon--right"
  />
</div>
`;
